pallets/frequency-tx-payment/src/tests.rs

```diff
@@ -664,3 +698,70 @@ fn compute_capacity_fee_successful() {
 			assert_eq!(fee, 26);
 		});
 }
+
+#[test]
+fn pay_with_capacity_batch_all_errors_when_transaction_amount_exceeds_maximum() {
+	let balance_factor = 10;
+	let too_many_calls = vec![
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+	];
+
+	ExtBuilder::default()
+		.balance_factor(balance_factor)
+		.base_weight(Weight::from_ref_time(5))
+		.build()
+		.execute_with(|| {
+			let who = 1u64;
+
+			assert_noop!(
+				FrequencyTxPayment::pay_with_capacity_batch_all(
+					RuntimeOrigin::signed(who),
+					too_many_calls
+				),
+				Error::<Test>::BatchedCallAmountExceedMaximum
+			);
+		});
+}
+
+#[test]
+fn pay_with_capacity_batch_all_transactions_will_all_fail_if_one_fails() {
+	let balance_factor = 10;
+
+	ExtBuilder::default()
+		.balance_factor(balance_factor)
+		.base_weight(Weight::from_ref_time(5))
+		.build()
+		.execute_with(|| {
+			let origin = 1u64;
+			let successful_balance_transfer_call =
+				RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 });
+
+			let balance_transfer_call_insufficient_funds =
+				RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100000000 });
+
+			let token_balance_before_call = Balances::free_balance(origin);
+
+			let calls_to_batch =
+				vec![successful_balance_transfer_call, balance_transfer_call_insufficient_funds];
+
+			let result = FrequencyTxPayment::pay_with_capacity_batch_all(
+				RuntimeOrigin::signed(origin),
+				calls_to_batch,
+			);
+
+			assert!(match result {
```

Comment:
what whats the Error you get when you fail?

assert_eq!(
  FrequencyTxPayment::pay_with_capacity_batch_all(
				RuntimeOrigin::signed(origin),
				calls_to_batch,
			),
			Err(DispatchErrorWithPostInfo {
				error: pallet_utility::<T>::Error::<Test>::SomeError,
				post_info: PostDispatchInfo {
					actual_weight: Some(....),
					pays_fee: Default::default(),
				},
			})
	);

Reply:
Ahh, I think it might be the error from BalancesCall.

Reply:
It is the `InsufficientBalance` error, however it is returned buried in the `DispatchErrorWithPostInfo`

Reply:
Maybe we can make it more specific later if necessary? I wanted to be able to capture the fact that the batch failed and that the state of the origin's balance has not change